src/content/content.js

```diff
@@ -6,12 +6,75 @@ import StateMessage from '@/lib/messaging/message/state-message'
 import StateResponse from '../lib/messaging/response/state-response'
 import MessagingService from '@/lib/messaging/service.js'
 import BgAuthenticator from '@/lib/auth/bg-authenticator.js'
-import { TabScript, UIController, ExtensionSyncStorage, HTMLPage, L10n, Locales, enUS, enGB } from 'alpheios-components'
+import { TabScript, UIController, UIEventController, ExtensionSyncStorage, HTMLPage, L10n, Locales, enUS, enGB,
+  LexicalQuery, ResourceQuery, AnnotationQuery, L10nModule, AuthModule, PanelModule, PopupModule,
+  MouseDblClick, LongTap, GenericEvt } from 'alpheios-components'
 import ComponentStyles from '../../node_modules/alpheios-components/dist/style/style.min.css' // eslint-disable-line
 
 let messagingService = null
 let uiController = null
 
+/*
+A UI controller's builder function customized for webextension
+ */
+const createUiController = (state, options) => {
```

Comment:
seems like we're going back and forth here over whether we can have a general purpose create method inside the UIController or not.  I don't really like the code duplication here across content, content-safari and embed-lib. Is the only difference across these 3 the way the auth module is registered?

Reply:
Well, I'm not sure how to best deal with it. Right now we have three UI controller owners: embed-lib, webextension, and Safari app extension. Their configuration would usually differ (but might be the same by accident). So where is the best place to put a builder funciton's code?

We could do that in a UI controller, but then it will be separated from the owner's code base. If webextension wants to change some initialization options, it has to do so inside the components lib which is not right.

So I thought that probably an idea to put a builder function's code into each owner's file would make sense. I've made them regular functions, not static members, to signify that this code is not UI controller specific, but related to an owner rather. But it's just a syntactic difference of cause.

What do you think? How would we handle it the best way?

Reply:
I am not sure.  The repetition of the registering of listeners and event handler subscriptions is what is worrying me most right now. It seems it could be very easy to forget one and miss behavior in the application as a result. Maybe those could be broken out into separate UIController methods, and then those methods could be called in the create code in the webextension and embed-lib? That way they can be overridden there only if necessary?

Reply:
Yes, that's a good idea. I will think how it's better to implement it. It does have duplicated code so far so I can probably move it back into a UI controller's method.